### PR TITLE
[5.5] Fix signature mismatch on _taskGroupAttachChild

### PR DESCRIPTION
**Pull request by kateinoigakukun**
This change was taken from ba615029c7c9413d848555e9517036a44acc48b2 via:

```
git checkout ba615029c7c9413d848555e9517036a44acc48b2 stdlib/public/Concurrency/TaskGroup.swift
```

Function signature consistency is very important on Wasm architecture, so cherry-picked 🍒 

CC: @ktoso

### CODE REVIEW

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Review by ktoso — APPROVED** [review]
LGTM, thanks. Yeah the function is void returning in C++ if I remember correctly, so this looks correct here.

Thanks!

**Comment by ktoso** [comment]
@swift-ci please test

**Comment by MaxDesiatov** [comment]
Do we need someone else to merge, since this is a release branch?

**Comment by ktoso** [comment]
Yeah; I've reached out internally to check if this is ok and if we can merge.

**Comment by ktoso** [comment]
Okey looping back here after a while -- sorry for the wait.

--- 

So this has been since addressed in a different way. The "attach" happens nowadays in happens inside spawning tasks so the  `_taskGroupAttachChild` is gone entirely now.

I'm sorry about the "internal" phrasing by the way. The process we use to manage and merge changes to 5.5 is completely public and documented on the forums: https://forums.swift.org/t/swift-5-5-release-process/45644 Specifically the notes on _Pull Requests for Release Branch_ explain the process here - so we would have needed a code owners approval here.

I'll do better about following and pointing out the process next time; and thanks for your time and PR!

**Pull request closed without merging**
